relayer/src/chain/cosmos/events.rs

```diff
@@ -0,0 +1,59 @@
+//! Extension events specific to chains followed via Cosmos SDK.
+
+use serde_derive::Serialize;
+
+use tendermint::abci::tag::Tag;
+use tendermint::abci::Event as AbciEvent;
+
+use ibc::Height;
+
+// Event type strings
+const SUBMIT_PROPOSAL_EVENT: &str = "submit_proposal";
+
+/// Variants that encode the possible message types sent via Cosmos SDK modules
+#[derive(Debug, Serialize)]
+pub enum CosmosEvent {
+    /// An event sent via the governance module indicating a new governance proposal was submitted
+    SubmitProposal(SubmitProposal),
+
+    ChainError(String), // Special event, signifying an error on CheckTx or DeliverTx
+}
+
+impl CosmosEvent {
+    pub fn try_from_tx_response(height: Height, event: &AbciEvent) -> Option<CosmosEvent> {
+        match &event.type_str[..] {
+            SUBMIT_PROPOSAL_EVENT => {
+                SubmitProposal::try_from_height_event_tags(height, &event.attributes)
+                    .map(CosmosEvent::SubmitProposal)
+            }
+            _ => None,
+        }
+    }
+}
+
+/// Data of a "submit_proposal" event carrying a "proposal_id" tag.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
+pub struct SubmitProposal {
+    pub proposal_id: u64,
```

Comment:
Let's introduce a new type for this?

Reply:
I'd rather not overdo it at this point, since we don't have any use of the field besides dumping it in the command output.